### PR TITLE
gh-124613: Deactivate perf support in tests if the jit is set

### PR DESCRIPTION
**Pull request by pablogsal**
* Issue: gh-124613

### CODE REVIEW

**Comment by pablogsal** [comment]
CC @brandtbucher @diegorusso

**Comment by pablogsal** [comment]
Hummm if we are going to make the JIT the default maybe we should only deactivate in YES and not in Yes-off @brandtbucher WDYT?

**Comment by brandtbucher** [comment]
> Hummm if we are going to make the JIT the default maybe we should only deactivate in YES and not in Yes-off @brandtbucher WDYT?

I mean, a `yes-off` build can still have a JIT if the env var is set (ditto for unsetting it via an env var on `yes` builds). We really just want to deactivate the JIT, regardless of which mode it was built in or what env vars are set.

Check out `test.support.without_optimizer`, I think it's probably what you want here.

**Comment by brandtbucher** [comment]
Oh, whoops, wrong PR. I thought this was the one that's skipping perf tests.

**Comment by brandtbucher** [comment]
Yeah, we could just say that both `yes` and `yes-off` disable perf jit support. Your call... but yeah, that means we can't ship perf JIT support in the same build if we do that.

Perhaps we should just disable the JIT if perf support is activated at runtime?

**Comment by pablogsal** [comment]
> Yeah, we could just say that both `yes` and `yes-off` disable perf jit support. Your call... but yeah, that means we can't ship perf JIT support in the same build if we do that.
> 
> Perhaps we should just disable the JIT if perf support is activated at runtime? It requires a bit more surgery, but I think it's probably the best option.

Yeah we should use both things I think. Currently it's not switched to `yes-off` no?

**Comment by brandtbucher** [comment]
> Currently it's not switched to `yes-off` no?

That's too many negatives for me to follow, haha.

What are you proposing, exactly? Disable the JIT when perf support is turned on at runtime? I believe this needs to cover `PYTHON_PERF_JIT_SUPPORT`, `PYTHONPERFSUPPORT`, `-X perf`, and `-X perf_jit` at startup, and `sys.activate_stack_trampoline()` and `sys.deactivate_stack_trampoline()` at runtime.

**Comment by pablogsal** [comment]
> What are you proposing, exactly? Disable the JIT when perf support is turned on at runtime? 

Yes, but that in a separate PR

For now I have modified this PR to use the `test.support.without_optimizer` so this always runs without the JIT. Is there a way to do that with an env var instead? That way I can call the subprocess without the JIT always

**Comment by brandtbucher** [comment]
> For now I have modified this PR to use the `test.support.without_optimizer` so this always runs without the JIT. Is there a way to do that with an env var instead? That way I can call the subprocess without the JIT always

`PYTHON_JIT=0`

**Comment by pablogsal** [comment]
> > For now I have modified this PR to use the `test.support.without_optimizer` so this always runs without the JIT. Is there a way to do that with an env var instead? That way I can call the subprocess without the JIT always
> 
> `PYTHON_JIT=0`

Ok, moved this PR to that

File: Lib/test/test_perf_profiler.py [hunk 1]
```diff
@@ -71,11 +62,13 @@ def baz():
                 """
         with temp_dir() as script_dir:
             script = make_script(script_dir, "perftest", code)
+            env = {**.os.getenv(), "PYTHON_JIT": "0"}
```
Comment (pablogsal) [inline Lib/test/test_perf_profiler.py]:
```suggestion
            env = {**os.getenv(), "PYTHON_JIT": "0"}
```

File: Lib/test/test_perf_profiler.py [hunk 2]
```diff
@@ -139,11 +132,13 @@ def baz():
...
```
Comment (pablogsal) [inline Lib/test/test_perf_profiler.py]:
```suggestion
            env = {**os.getenv(), "PYTHON_JIT": "0"}
```

File: Lib/test/test_perf_profiler.py [hunk 3]
```diff
@@ -188,11 +183,13 @@ def baz():
...
```
Comment (pablogsal) [inline Lib/test/test_perf_profiler.py]:
```suggestion
            env = {**os.getenv(), "PYTHON_JIT": "0"}
```

File: Lib/test/test_perf_profiler.py [hunk 4]
```diff
@@ -71,11 +62,13 @@ def baz():
                 """
         with temp_dir() as script_dir:
             script = make_script(script_dir, "perftest", code)
+            env = {**os.getenv(), "PYTHON_JIT": "0"}
```
Comment (diegorusso) [inline Lib/test/test_perf_profiler.py]: `os.getenv()` wants a key for environment variable to get. I think what you want is

```
env =  {**os.environ, "PYTHON_JIT": "0"}
```

Comment (pablogsal) [reply Lib/test/test_perf_profiler.py]: Lol yes 🤦 

man the jet lag is not good 😓

**Review by brandtbucher — APPROVED** [review]

**Pull request merged**